src/asm2wasm.h

```diff
@@ -3229,6 +3229,10 @@ Function* Asm2WasmBuilder::processFunction(Ref ast) {
   // remove that shift. if there is a shift, we can just look through it, etc.
   processUnshifted = [&](Ref ptr, unsigned bytes) {
     auto shifts = bytesToShift(bytes);
+    // E.g. the address variable "$4" in Atomics_compareExchange(HEAP8, $4, $7, $8);
+    if (ptr->isString()) {
+      return process(ptr);
+    }
```

Comment:
Hmm, the function, as the comment says, is called for stuff like `HEAP32[addr >> 2]`. So if it's called on `Atomics_compareExchange(.., $4)` which does *not* have a shift, then that call needs to be changed I think.

Reply:
I'm not sure what you mean. Do you mean we should change fastcomp to emit `Atomics_compareExchange(HEAP8>>0, $4, $7, $8);` instead of `Atomics_compareExchange(HEAP8, $4, $7, $8);`?

Or do you mean that this function `processUnshifted` should not be called if there is no shift? Having to detect the lack of shift in all places before calling that function would be quite tedious/tricky?

Reply:
Sorry, I meant that the caller here (not in fastcomp) should be fixed. That is, the place (looks like there might be more than 1) in which we detect an `Atomics_compareExchange` should not be calling `processUnshifted`, it should instead do what the fix here says to do.

Otherwise, this changes processUnshifted for all callers, which seems risky for bugs.

Reply:
Ok, adjusted the fix to only apply to `processIgnoringShift`. How does that look now?